sdks/go/pkg/beam/transforms/xlang/inference/inference.go

```diff
@@ -0,0 +1,115 @@
+	beam.RegisterType(reflect.TypeOf((*argsStruct)(nil)).Elem())
+	beam.RegisterType(reflect.TypeOf((*sklearnKwargs)(nil)).Elem())
+	beam.RegisterType(reflect.TypeOf((*PredictionResult)(nil)).Elem())
+}
+
+var outputT = reflect.TypeOf((*PredictionResult)(nil)).Elem()
+
+// PredictionResult represents the result of a prediction obtained from Python's RunInference API.
+type PredictionResult struct {
+	Example   []int64 `beam:"example"`
+	Inference int32   `beam:"inference"`
+}
+
+type sklearnConfig struct {
+	kwargs        sklearnKwargs
+	args          argsStruct
+	expansionAddr string
+}
+
+type sklearnConfigOption func(*sklearnConfig)
+
+// WithArgs set arguments for the sklearn inference transform parameters
+func WithArgs(args []string) sklearnConfigOption {
+	return func(c *sklearnConfig) {
+		c.args.args = append(c.args.args, args...)
+	}
+}
+
+// WithExpansionAddr provides URL for Python expansion service.
+func WithExpansionAddr(expansionAddr string) sklearnConfigOption {
+	return func(c *sklearnConfig) {
+		c.expansionAddr = expansionAddr
+	}
+}
+
+type argsStruct struct {
+	args []string
+}
+
+// sklearnKwargs defines acceptable keyword args for Sklearn Model Handler.
+type sklearnKwargs struct {
+	// ModelHandlerProvider defines the model handler to be used.
+	ModelHandlerProvider python.CallableSource `beam:"model_handler_provider"`
+	// ModelURI indicates the model path to be used for Sklearn Model Handler.
+	ModelURI string `beam:"model_uri"`
+}
+
+// Sklearn provides inference over a SklearnModelHandler.
+// ModelURI is the required parameter indicating the path to the sklearn model.
+// This wrapper doesn't work for keyed input PCollection.
+//
+// Example:
+//
+//		inputRow := [][]int64{{0, 0}, {1, 1}}
+//	    input := beam.CreateList(s, inputRow)
+//	    modelURI = gs://example.com/tmp/staged/sklearn_model
+//		predictions := inference.Sklearn(s, modelURI, input, inference.WithExpansionAddr(expansionAddr))
+func Sklearn(s beam.Scope, modelUri string, col beam.PCollection, opts ...sklearnConfigOption) beam.PCollection {
+	s.Scope("xlang.inference.Sklearn")
+
+	cfg := sklearnConfig{}
+	for _, opt := range opts {
+		opt(&cfg)
+	}
+	cfg.kwargs.ModelHandlerProvider = python.CallableSource("apache_beam.ml.inference.sklearn_inference.SklearnModelHandlerNumpy")
+	cfg.kwargs.ModelURI = modelUri
+	return runInference[sklearnKwargs](s, col, cfg.args, cfg.kwargs, cfg.expansionAddr)
+}
+
+func runInference[Kwargs any](s beam.Scope, col beam.PCollection, a argsStruct, k Kwargs, addr string) beam.PCollection {
+	expansionAddr := addr
+	if expansionAddr == "" {
+		expansionAddr = xlangx.UseAutomatedPythonExpansionService(python.ExpansionServiceModule)
+	}
+	pet := python.NewExternalTransform[argsStruct, Kwargs]("apache_beam.ml.inference.base.RunInference.from_callable")
+	pet.WithKwargs(k)
+	pet.WithArgs(a)
+	pl := beam.CrossLanguagePayload(pet)
+	namedInput := map[string]beam.PCollection{xlang.SetOutputCoder: col}
```

Comment:
It's very odd that the output coder is being added to the inputs, rather than the outputs?

As discussed, I thought we were going to add this as an alternative to the "unamed outputs" and then translate that to unnamed downstream.